packages/@internationalized/date/tests/manipulation.test.js

```diff
@@ -226,9 +226,9 @@ describe('CalendarDate manipulation', function () {
         expect(date.subtract({months: 1})).toEqual(new CalendarDate(new PersianCalendar(), 1, 1, 1));
       });
 
-      it('should constrain when reaching year 9377', function () {
-        let date = new CalendarDate(new PersianCalendar(), 9377, 12, 10);
-        expect(date.add({months: 1})).toEqual(new CalendarDate(new PersianCalendar(), 9377, 12, 31));
+      it('should constrain when reaching year 3178', function () {
+        let date = new CalendarDate(new PersianCalendar(), 3178, 12, 10);
+        expect(date.add({months: 1})).toEqual(new CalendarDate(new PersianCalendar(), 3178, 12, 31));
```

Comment:
I'm not sure this is quite right. We generally want to match the output that Temporal, the upcoming JS standard, produces. If you try on https://tc39.es/proposal-temporal/docs/plaindate.html:

```js
let date = Temporal.PlainDate.from({year: 3178, month: 12, day: 10, calendar: 'persian'}).add({months: 1});
date.year // => 3179
date.month // => 1
date.day // => 10
```

That date converts to 3780-03-29 Gregorian, which formats using Intl.DateTimeFormat as a valid date in the Persian calendar.

```js
Intl.DateTimeFormat('en-US-u-ca-persian').format(new Date(3780, 3 - 1, 29))
// => '1/10/3159 AP'
```

I guess this was due to the change in `getYearsInEra`, which results in the date being constrained to this maximum. Temporal and Intl.DateTimeFormat don't seem to have this limit so I'm not sure we should either.